hi-ml-cpath/src/health_cpath/utils/plots_utils.py

```diff
@@ -216,6 +234,28 @@ def get_list_from_results_dict(results: ResultsType, results_key: ResultsKey) ->
     return [i.item() if isinstance(i, Tensor) else i for i in results[results_key]]
 
 
+def get_stratified_outputs(true_labels: List[Any], scores: List[Any],
+                           stratify_metadata: List[Any]) -> Dict[str, List[Any]]:
+    """
+    Get stratified true labels and predictions given metadata, from all true and prediction labels
+    to plot stratified curves.
+    :param true_labels: list of true labels.
+    :param scores: list of prediction scores.
+    :param stratify_metadata: list containing the corresponding metadata values on which to stratify results.
+    """
+    unique_vals, unique_counts = np.unique(stratify_metadata, return_counts=True)   # metadata should not contain nans
+    stratified_outputs = {}
+    for i in range(len(unique_vals)):
+        val = unique_vals[i]
+        count = unique_counts[i]
+        idxs = [i for i, x in enumerate(stratify_metadata) if x == val]
```

Comment:
Not sure this will work, but can we do something like `np.argwhere` here?

Reply:
https://numpy.org/doc/stable/reference/generated/numpy.argwhere.html from the documentations, it seems that it returns Indices of elements "that are non-zero". I'll skip this for now, thanks